dask/array/tests/test_random.py

```diff
@@ -161,6 +161,11 @@ def test_random_all():
     da.random.standard_normal(size=5, chunks=3).compute()
     da.random.standard_t(2, size=5, chunks=3).compute()
 
+def test_array_broadcasting():
+    arr = np.arange(6).reshape((2,3))
+    assert da.random.poisson(arr, chunks=3).compute().shape == (2,3)
+    assert da.random.normal(arr, 2, chunks=3).compute().shape == (2,3)
```

Comment:
It seems like the options for broadcasting are pretty rich here.  We should probably have tests that cover all of the classes of cases that we might encounter.  

What happens if a user also provides a size?  
What happens if there are multiple arrays that do or do not share the same shape?  
What happens if the chunkshape doesn't match?